app/javascript/components/patient/follow_up_flag/FollowUpFlag.js

```diff
@@ -0,0 +1,268 @@
+        let idArray = this.props.patients.map(x => x['id']);
+        axios
+          .post(window.BASE_PATH + '/patients/bulk_edit', {
+            ids: idArray,
+            bulk_edit_type: 'follow-up',
+            follow_up_reason: this.state.follow_up_reason,
+            follow_up_note: this.state.follow_up_note,
+            clear_flag: this.state.clear_flag,
+            clear_flag_reason: this.state.clear_flag_reason,
+            apply_to_household: this.state.bulk_action_apply_to_household,
+          })
+          .then(() => {
+            location.href = window.BASE_PATH;
+          })
+          .catch(error => {
+            reportError('Failed to bulk update Follow-up Flag.');
+            console.log(error);
+          });
+      } else {
+        axios
+          .post(window.BASE_PATH + '/patients/' + this.props.patient.id + '/follow_up_flag', {
+            follow_up_reason: this.state.follow_up_reason,
+            follow_up_note: this.state.follow_up_note,
+            clear_flag: this.state.clear_flag,
+            clear_flag_reason: this.state.clear_flag_reason,
+            apply_to_household: this.state.apply_to_household,
+            apply_to_household_ids: this.state.apply_to_household_ids,
+          })
+          .then(() => {
+            // Reload the page to see the flag on the monitoree's page
+            location.reload();
+          })
+          .catch(error => {
+            reportError(error?.response?.data?.error ? error.response.data.error : error, false);
+            console.log(error);
+          });
+      }
+    });
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <Modal.Body>
+          <Form.Group style={{ display: 'inline-flex' }}>
+            <Form.Check
+              type="radio"
+              name="flag_for_follow_up_option"
+              className="pr-5"
+              id="set_flag_for_follow_up"
+              label="Set Follow-up Flag"
+              onChange={this.handleChange}
+              checked={!this.state.clear_flag}
+            />
+            <Form.Check
+              type="radio"
+              name="flag_for_follow_up_option"
+              id="clear_flag_for_follow_up"
+              label="Clear Follow-up Flag"
+              disabled={this.state.clear_flag_disabled}
+              onChange={this.handleChange}
+              checked={this.state.clear_flag}
+            />
+          </Form.Group>
+          {!this.state.clear_flag && (
+            <React.Fragment>
+              <Form.Group controlId="follow_up_reason">
+                <Form.Label>
+                  Please select a reason for being flagged for follow-up. If a monitoree is already flagged, this reason will replace any previously selected
+                  reason.
+                </Form.Label>
+                <Form.Control as="select" size="lg" className="form-square mb-3" value={this.state.follow_up_reason} onChange={this.handleChange}>
+                  <option></option>
```

Comment:
Should we be validating these options in the patient model? Right now the monitoring reason can be set to any arbitrary value

Reply:
If so could we construct the options similar to [here](https://github.com/SaraAlert/SaraAlert/blob/c894427c2997239df81246a57ea148250e6a132a/app/javascript/components/patient/monitoring_actions/CaseStatus.js#L224) so that we only need to keep track of the valid options in one place?

Reply:
@jkufro she originally had it that way, but I requested to change it because it is only in this modal.  before it was passing the list of options as props through like 8 different components.  if a list of options is used once in the UI we typically just hardcode it there